src/tools/illink/src/linker/Linker.Steps/MarkStep.MethodDefinitionDependencyNode.cs

```diff
@@ -0,0 +1,83 @@
+		/// </summary>
+		internal sealed class PostPoneMethodProcessingNode : DependencyNodeCore<MarkStepNodeFactory>
+		{
+			readonly MethodDefinition method;
+			readonly DependencyInfo reason;
+			readonly MessageOrigin origin;
+
+			public PostPoneMethodProcessingNode (MethodDefinition method, DependencyInfo reason, MessageOrigin origin)
+			{
+				this.method = method;
+				this.reason = reason;
+				this.origin = origin;
+			}
+
+			public override bool InterestingForDynamicDependencyAnalysis => false;
+
+			public override bool HasDynamicDependencies => false;
+
+			public override bool HasConditionalStaticDependencies => false;
+
+			public override bool StaticDependenciesAreComputed => true;
+
+			public override IEnumerable<DependencyListEntry>? GetStaticDependencies (MarkStepNodeFactory context)
+			{
+				yield return new DependencyListEntry(context.MarkStep.GetMethodDefinitionNode (method, reason, origin), "Needed by dummy node");
+			}
+
+			public override IEnumerable<CombinedDependencyListEntry>? GetConditionalStaticDependencies (MarkStepNodeFactory context) => null;
+			public override IEnumerable<CombinedDependencyListEntry>? SearchDynamicDependencies (List<DependencyNodeCore<MarkStepNodeFactory>> markedNodes, int firstNode, MarkStepNodeFactory context) => null;
+			protected override string GetName (MarkStepNodeFactory context) => "PostPoneMethodMarkingProxy";
+		}
+
+		internal sealed class MethodDefinitionDependencyNode : DependencyNodeCore<MarkStepNodeFactory>
+		{
+			readonly MethodDefinition method;
+			readonly MessageOrigin origin;
+			readonly DependencyInfo reason;
+
+			public MethodDefinitionDependencyNode (MethodDefinition method, DependencyInfo reason, MessageOrigin origin)
+			{
+				this.method = method;
+				this.origin = origin;
+				this.reason = reason;
+			}
+
+			public override bool InterestingForDynamicDependencyAnalysis => false;
+
+			public override bool HasDynamicDependencies => false;
+
+			public override bool HasConditionalStaticDependencies => false;
+
+			public override bool StaticDependenciesAreComputed => true;
+
+			public override IEnumerable<DependencyListEntry>? GetStaticDependencies (MarkStepNodeFactory context)
+			{
+				// Add other types that are marked in MarkType
+				yield break;
+			}
+
+			public override IEnumerable<CombinedDependencyListEntry>? GetConditionalStaticDependencies (MarkStepNodeFactory context) => null;
+			public override IEnumerable<CombinedDependencyListEntry>? SearchDynamicDependencies (List<DependencyNodeCore<MarkStepNodeFactory>> markedNodes, int firstNode, MarkStepNodeFactory context) => null;
+			protected override string GetName (MarkStepNodeFactory context) => "MethodDefinition";
+			protected override void OnMarked (MarkStepNodeFactory context)
+			{
+				context.MarkStep.ProcessMethod (method, reason, origin);
```

Comment:
Given that you want this to be postponed anyway - why not call it from GetStaticDependencies - that will postpone it and you won't need the dummy node above to do that for you.

Reply:
Nice, I didn't realize that would have the same effect. Moved the call to GetStaticDependencies.